### PR TITLE
[codegen/nodejs] Implement Lazy Loading

### PR DESCRIPTION
**Pull request by RobbieMcKinstry**
# Description

This PR introduces lazy-loading into NodeJS codegen by dynamically `requiring` modules instead of eagerly importing them.

The generated syntax isn't quite right at this time. It needs another pass for correctness.

Fixes https://github.com/pulumi/pulumi/issues/7653

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] I have updated the [CHANGELOG-PENDING](https://github.com/pulumi/pulumi/blob/master/CHANGELOG_PENDING.md) file with my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

**Comment by t0yv0** [comment]
Exciting to see this through! Looks like a fee more pieces missing like doing genLazyLoad call and fixing up some codegen tests, but as we discussed before, the approach seems very promising!

File: pkg/codegen/nodejs/gen.go [hunk 1]
```diff
@@ -2559,7 +2552,21 @@ export function getVersion(): string {
 export function resourceOptsDefaults(): any {
     return { version: getVersion()%s };
 }
-`
+
+/** @internal */
+function lazy_load(exports: any, module_name: string) {
```
[the remark this answers is not in the document]
Comment (RobbieMcKinstry) [reply pkg/codegen/nodejs/gen.go:2557]: Indeed it does! lol

**Comment by t0yv0** [comment]
Went with https://github.com/pulumi/pulumi/pull/10538

**Comment by RobbieMcKinstry** [comment]
> Do you know how to accept codgen tests? As in, `cd pkg/codegen/nodejs && PULUMI_ACCEPT=1 go test`. This will rewrite expected files to actual automatically, and we can review.

I had encountered that before, but I wouldn't have been able to recall it if under the gun. Thanks for the reminder! Going to save that in my lab notebook.

**Pull request closed without merging**